org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -237,10 +249,20 @@ public void initialize() {
         // If no key exists, generate a random key and save it back to the configuration
         if (!networkKey.isValid()) {
             networkKey = ZigBeeKey.createRandom();
-            logger.debug("Key initialised {}", networkKey);
+            logger.debug("Network key initialised {}", networkKey);
         }
 
-        logger.debug("Key final array {}", networkKey);
+        logger.debug("Network key final array {}", networkKey);
+
+        // Process the link key
+        try {
+            logger.debug("Link Key String {}", linkKeyString);
+            linkKey = new ZigBeeKey(linkKeyString);
+        } catch (IllegalArgumentException e) {
```

Comment:
Minor suggestion: Add a log statement within the `catch` block, so that it gets visible in the log that (and why) the system falls back to the well-known key.

That the system fell back to the well-known key is also visible (though somewhat implicit) from the log statement with the final link key array, the additional log statement would make this more explicit.